platform/android/src/jni.cpp

```diff
@@ -1515,10 +1515,12 @@ void setOfflineRegionObserver(JNIEnv *env, jni::jobject* offlineRegion_, jni::jo
             }
 
             // Error object
-            jni::jobject* jerror = &jni::NewObject(*env2, *offlineRegionErrorClass, *offlineRegionErrorConstructorId);
```

Comment:
Ok, well, I'm really skeptical about switching everything from `PushLocalFrame`/`PopLocalFrame` to `DeleteLocalRef`. My hunch:

* `PushLocalFrame` throwing `OutOfMemoryError` is a symptom of local references leaking elsewhere. It just happens to be the place where hit the ceiling.
* Switching to `DeleteLocalRef` happens to prevent hitting the ceiling, because allocations/deallocations are slightly more granular, but doesn't fix the underlying leak.